rewrite-protobuf/src/main/java/org/openrewrite/protobuf/ProtoParser.java

```diff
@@ -0,0 +1,102 @@
+import org.antlr.v4.runtime.*;
+import org.intellij.lang.annotations.Language;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.Parser;
+import org.openrewrite.internal.MetricsHelper;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.protobuf.internal.ProtoParserVisitor;
+import org.openrewrite.protobuf.internal.grammar.Protobuf2Lexer;
+import org.openrewrite.protobuf.internal.grammar.Protobuf2Parser;
+import org.openrewrite.protobuf.tree.Proto;
+import org.openrewrite.tree.ParsingEventListener;
+import org.openrewrite.tree.ParsingExecutionContextView;
+
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.stream.Collectors.toList;
+
+public class ProtoParser implements Parser<Proto.Document> {
+    @Override
+    public List<Proto.Document> parseInputs(Iterable<Input> sourceFiles, @Nullable Path relativeTo, ExecutionContext ctx) {
+        ParsingEventListener parsingListener = ParsingExecutionContextView.view(ctx).getParsingListener();
+        return acceptedInputs(sourceFiles).stream()
+                .map(sourceFile -> {
+                    Timer.Builder timer = Timer.builder("rewrite.parse")
+                            .description("The time spent parsing an Json file")
+                            .tag("file.type", "Json");
+                    Timer.Sample sample = Timer.start();
+                    try (InputStream sourceStream = sourceFile.getSource()) {
+                        Protobuf2Parser parser = new Protobuf2Parser(new CommonTokenStream(new Protobuf2Lexer(
+                                CharStreams.fromStream(sourceStream))));
+
+                        parser.removeErrorListeners();
+                        parser.addErrorListener(new ForwardingErrorListener(sourceFile.getPath(), ctx));
+
+                        Proto.Document document = new ProtoParserVisitor(
+                                sourceFile.getRelativePath(relativeTo),
+                                StringUtils.readFully(sourceFile.getSource())
+                        ).visitProto(parser.proto());
+                        sample.stop(MetricsHelper.successTags(timer).register(Metrics.globalRegistry));
+                        parsingListener.parsed(sourceFile, document);
+                        return document;
+                    } catch (Throwable t) {
+                        sample.stop(MetricsHelper.errorTags(timer, t).register(Metrics.globalRegistry));
+                        ctx.getOnError().accept(new IllegalStateException(sourceFile.getPath() + " " + t.getMessage(), t));
+                        return null;
+                    }
+                })
+                .filter(Objects::nonNull)
+                .collect(toList());
+    }
+
+    @Override
+    public List<Proto.Document> parse(@Language("Json") String... sources) {
```

Comment:
```suggestion
    public List<Proto.Document> parse(String... sources) {
```
I checked and IntelliJ doesn't have any built-in language injection for protobuf